tmt/base.py

```diff
@@ -496,6 +497,21 @@ def __init__(self, node, run=None):
             (key, str(value)) for key, value
             in node.get('environment', dict()).items()])
 
+        # Environment variables from files
+        env_files = self.node.get("environment-file") or []
+        assert isinstance(env_files, list), f"environment_file parameter should be a list. " \
+                                            f"Received {type(env_files)}"
+        env_files_vars = tmt.utils.environment_file_to_dict(env_files)
+        assert not set(env_files_vars).intersection(set(self._environment)), (
+            "Variables sets in environment and environment_file are conflicting."
+            )
+        self._environment.update(tmt.utils.environment_file_to_dict(env_files))
+
+        # ensure that presented vars in os.environment was not overwritten
+        # and revert these values if present
+        for var in filter(lambda var_: var_ in os.environ, self._environment):
```

Comment:
Instead of filter using `[var for var in self._environment if var in os.environ]` is more consistent with the rest of tmt code.

Reply:
okay